packages/medusa/src/api/routes/store/products/index.ts

```diff
@@ -124,19 +126,25 @@ export type StoreProductsRes = {
 /**
  * @schema StorePostSearchRes
  * type: object
+ * required:
+ *   - hits
  * properties:
  *   hits:
- *     type: array
  *     description: Array of results. The format of the items depends on the search engine installed on the server.
+ *     type: array
  */
 export type StorePostSearchRes = {
   hits: unknown[]
-  [k: string]: unknown
```

Comment:
note: I could not find usage or example of this field in the codebase.

Reply:
The type was added as we cannot control the response from installed search services. We require `hits`, but usually, they return additional properties specific to the provider.

As an example, the SS below is a search hit from an Algolia installation:

![CleanShot 2023-02-17 at 09 46 54](https://user-images.githubusercontent.com/59018053/219597084-275ef5b6-505b-460d-90a8-f8abb27e1646.png)

Reply:
Make sense. I re-introduced the support for unknown fields. The typing is different but it achieves the same result and I can more easily replicate it in OAS. See 2b8779e9abf6ee11ff6decf79c48b53d8d429390